docs/rfc/0000-cli-infrastructure.adoc

```diff
@@ -0,0 +1,254 @@
+
+The author recommends *2.*. The modularity provides a clean separation
+of concerns and avoids a monolothic approach. We assume good faith
+of overrides not happening [TODO: is there some way we can just
+disallow this programatically?]. For packaging and distribution of the
+binaries, we propose to have a script that will install each
+subcommand binary — until we have a more well-founded way of
+distributing Radicle artifacts.
+
+=== `radicle-bins`
+
+Per RFC-682#CLI, all subcommands will be provided as
+libraries. However, this leaves the question of where the binary might
+live. So far, it has been an unspoken formalism that all binaries live
+in `radicle-bins` where we can provide a `Cargo.lock` file. All
+functionality will live in `radicle-link`, including the `main`
+function, and the entry in `radicle-bins` will be a thin wrapper
+around this function.
+
+=== Distribution
+
+We are noting here that -- at the time of writing -- we cannot publish
+these binaries to <<crates.io>> due to the fact that `librad` and other
+crucial libraries only live as a `git` repositories and are not on
+crates.io. Anything that depends on them cannot be published as a
+consequence.
+
+This means that all subcommands will need to be built from source
+initially, unless we build the binaries and distribute them from a
+well-known source.
+
+== Core
+
+With all the details in place, we define a non-exhaustive list of the
+core commands that we wish to include for the first iteration. More
+components MAY be added as the result of future RFCs. We SHALL keep a
+list of the core commands as part of the `radicle-link` documentation
+— as part of the `README`.
+
+=== `rad-project`
+
+This command will provide management of the `Project` identities and
+their respective working copies. The following actions will be
+defined:
+
+* `create` — create a `Project` identity and either create a
+  respective working copy, or link it to an existing working copy.
+* `get` — attempt to retrieve a `Project` identity from the store based on
+  a provided URN.
+* `list` — list all `Project` identities in the store.
+* `update` — update a `Project` identity with a new payload.
+* `merge` — merge a peer's version of the `Project` identity with your own
+* `track` — track a peer for the `Project` identity and add the peer
+to the working copy's include file.
+
+=== `rad-person`
+
+This command will provide the management of the `Person`
+identities. The following actions will be defined:
+
+* `create`— create a `Person` identity.
+* `get`— attempt to retrieve a `Person` identity from the store based
+  on a provided URN.
+* `list`— list all `Person` identities in the store.
+* `update`— update a `Person` identity with a new payload.
+* `merge`— merge a peer's version of the `Person` identity with your own.
+* `track`— track a peer for the `Person` identity.
+
+=== `rad-replicate`
+
+* `ask` — ask the network for a given URN and replicate it to local storage.
```

Comment:
Those are two things.

Reply:
Would it be possible to track a URN and have the p2p stack take care of replicating it from the network?

Reply:
> Would it be possible to track a URN and have the p2p stack take care of replicating it from the network?

There's an issue for that actually =] https://github.com/radicle-dev/radicle-link/issues/141

Reply:
Contributions welcome. Essentially, we could just treat a remote without a `url` as a clone intent. Requires some work to decouple from libgit2